src/screens/SpeedUpAndCancelSheet.js

```diff
@@ -0,0 +1,405 @@
+import { useRoute } from '@react-navigation/native';
+import { captureException } from '@sentry/react-native';
+import { BigNumber } from 'bignumber.js';
+import { get, isEmpty } from 'lodash';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { InteractionManager, TurboModuleRegistry } from 'react-native';
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+} from 'react-native-reanimated';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components/native';
+import { GasSpeedButton } from '../components/gas';
+import { Centered, Column, Row } from '../components/layout';
+
+import {
+  SheetActionButton,
+  SheetActionButtonRow,
+  SheetHandleFixedToTop,
+  SlackSheet,
+} from '../components/sheet';
+import { Emoji, Text } from '../components/text';
+import { getTransaction, toHex } from '../handlers/web3';
+import TransactionStatusTypes from '../helpers/transactionStatusTypes';
+import { sendTransaction } from '../model/wallet';
+import { gweiToWei, weiToGwei } from '../parsers/gas';
+import { getTitle } from '../parsers/transactions';
+import { dataUpdateTransaction } from '../redux/data';
+import { updateGasPriceForSpeed } from '../redux/gas';
+import { safeAreaInsetValues } from '../utils';
+import deviceUtils from '../utils/deviceUtils';
+import {
+  useAccountSettings,
+  useDimensions,
+  useGas,
+  useKeyboardHeight,
+} from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import { colors, position } from '@rainbow-me/styles';
+import logger from 'logger';
+
+const isReanimatedAvailable = !(
+  !TurboModuleRegistry.get('NativeReanimated') &&
+  (!global.__reanimatedModuleProxy || global.__reanimatedModuleProxy.__shimmed)
+);
+
+const springConfig = {
+  damping: 500,
+  mass: 3,
+  stiffness: 1000,
+};
+
+const Container = styled(Centered).attrs({
+  direction: 'column',
+})`
+  ${position.cover};
+  ${({ deviceHeight, height }) =>
+    height ? `height: ${height + deviceHeight}` : null};
+`;
+
+const CenteredSheet = styled(Centered)`
+  border-top-left-radius: 39;
+  border-top-right-radius: 39;
+`;
+
+const AnimatedContainer = Animated.createAnimatedComponent(Container);
+const AnimatedSheet = Animated.createAnimatedComponent(CenteredSheet);
+
+const GasSpeedButtonContainer = styled(Row)`
+  justify-content: center;
+  margin-bottom: 19px;
+  width: ${deviceUtils.dimensions.width - 20};
+`;
+
+const CANCEL_TX = 'cancel';
+const SPEED_UP = 'speed_up';
+
+const title = {
+  [CANCEL_TX]: 'Cancel transaction',
+  [SPEED_UP]: 'Speed Up transaction',
```

Comment:
Should we do `Speed up transaction` to keep the casing more consistent

Reply:
@christianbaroni  I took it from the design